sdk/python/core/ydk/types/py_types.py

```diff
@@ -199,93 +202,106 @@ def get_child_by_name(self, child_yang_name, segment_path):
+        for name, leaf in self._leafs.items():
+            if leaf.name == path:
+                if isinstance(leaf, _YLeaf):
+                    if isinstance(self.__dict__[name], Bits):
+                        self.__dict__[name][value] = True
+                    else:
+                        self.__dict__[name] = value
+                elif isinstance(leaf, _YLeafList):
+                    self.__dict__[name].append(value)
 
     def set_filter(self, path, yfilter):
-        for _, val in vars(self).items():
-            if val is not None:
-                if isinstance(val, _YLeaf) or isinstance(val, _YLeafList):
-                    if _name_matches_yang_name(path, val.name):
-                        val.yfilter = yfilter
-                        return
+        pass
 
     def has_leaf_or_child_of_name(self, name):
-        for _, value in vars(self).items():
-            if value is not None:
-                if isinstance(value, _YLeaf) or isinstance(value, _YLeafList):
-                    if _name_matches_yang_name(name, value.name):
-                        return True
-        for path in self._child_list_classes:
-            if _name_matches_yang_name(name, path):
-                return True
-        for path in self._child_container_classes:
-            if _name_matches_yang_name(name, path):
+        for _, leaf in self._leafs.items():
+            if name == leaf.name:
                 return True
+
+        if self._child_list_classes.has_key(name):
+            return True
+
+        if self._child_container_classes.has_key(name):
+            return True
+
         return False
 
     def get_name_leaf_data(self):
         leaf_name_data = LeafDataList()
-        for _, value in vars(self).items():
-            if value is not None:
-                if isinstance(value, _YLeaf):
-                    if value.is_set or is_set(value.yfilter):
-                        leaf_name_data.append(value.get_name_leafdata())
-                elif isinstance(value, _YLeafList):
-                    leaf_name_data.extend(value.get_name_leafdata())
+        for name in self._leafs:
+            value = self.__dict__[name]
+            leaf = self._leafs[name]
+            if type(value) is _YFilter:
+                leaf.yfilter = value
+                if isinstance(leaf, _YLeaf):
+                    leaf_name_data.append(leaf.get_name_leafdata())
+                elif isinstance(leaf, _YLeafList):
+                    leaf_name_data.extend(leaf.get_name_leafdata())
+            elif (type(value) not in (list, type(None), Bits)
+                or (type(value) == Bits and len(value.get_bitmap()) > 0)):
+
+                leaf.set(value)
+                leaf_name_data.append(leaf.get_name_leafdata())
+            elif type(value) == list and len(value) > 0:
+                l = _YLeafList(YType.str, leaf.name)
+                # l = self._leafs[name]
+                # Above results in YPYModelError: 
```

Comment:
Based on these comments, is there anything pending to be done? Suggest add a `TODO` if so. Else, could the comment be removed?